deployment/terraform/assets/cluster.tf

```diff
@@ -34,8 +34,8 @@ resource "aws_instance" "app_server" {
   key_name      = aws_key_pair.key.id
   count         = var.app_instance_count
   vpc_security_group_ids = [
-    "${aws_security_group.app.id}",
-    "${aws_security_group.app_gossip.id}"
+    "${aws_security_group.app[0].id}",
+    "${aws_security_group.app_gossip[0].id}"
```

Comment:
This is somehow problematic and will fail with an "empty tuple error" in case of a single app server deployment. 
Not sure yet if making a bunch of `aws_security_group_rule` is the way to solve this case.

Reply:
Not sure if it's worth trying to optimize some spurious security group creations, if it's taking too much effort. Thoughts on leaving it as is?

Reply:
I tend to agree. Ideally I wanted to create only the strictly necessary but somehow I feel this is not the way to do it. Not sure if it's a limitation with terraform or just us using it not the intended way.

Reply:
Yeah, I don't think it's a limitation per-se. But security group dependencies are a bit tricky and time-consuming to get it right. If it's not a big issue, IMO there's no need to spend too much time on it and just let it be.